### DISCUSSION TITLE
How do i change the icon of a ui.button e.g. for a toggle button?

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [zauberzeug/nicegui#3596] [Ideas / Feature Requests]
### Question

I am trying
```python
def toggle_select_all_rows(self):
        """
        Toggle between selecting all rows and deselecting all rows.
        """
        if self.all_selected:
            self.deselect_all_rows()
        else:
            self.select_all_rows()
        self.all_selected=not self.all_selected
        self.update_select_all_toggle_button()

    def update_select_all_toggle_button(self):
        """
        Update the toggle button's label and icon based on selection state.
        """
        stb=self.select_toggle_button
        if self.all_selected:
            stb.text = "None"
            stb.icon = "select_off"
        else:
            stb.text = "All"
            stb.icon = "select_all"
        stb.update()
        
```

but the icon does not change when trying to implement https://github.com/WolfgangFahl/nicegui_widgets/issues/74 see https://ngdemo.bitplan.com/grid
      
      
<img width="157" alt="grafik" src="https://github.com/user-attachments/assets/63c49bcf-8d83-4d92-9016-9c40f04f1a1a">

<img width="720" alt="grafik" src="https://github.com/user-attachments/assets/bc8f1102-ea33-4072-8ee4-0b2fd0e6fed5">

**Comment by falkoschindler** [comment]
Currently there is no specific API for changing the icon of a button. But you can use the generic `props` method as a workaround:
```py
b = ui.button('Toggle', icon='check_box_outline_blank', on_click=lambda: b.props('icon=check_box'))
```

I'll convert this discussion into a feature request. We should add an `icon` property to the button class.
Even better: We could introduce an `IconElement` mixin with a bindable icon property. This could be used by `ui.avatar`, `ui.button_dropdown`, `ui.button`, `ui.chip`, `ui.expansion`, `ui.step`, `ui.tab`, and `ui.timeline`.

**Comment by WolfgangFahl** [comment]
<img width="115" alt="grafik" src="https://github.com/user-attachments/assets/1ccc3d7b-3c55-4d3c-bbaf-ec565e906d77">

**Comment by falkoschindler** [comment]
With the new `IconElement` mixin from PR #3600 being released in version 1.4.37, I'll  close this feature request as completed.
